src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,167 @@
+  ({ disabled, className, ...props }, ref) => {
+    const theme = useTheme()
+    return (
+      <Wrapper $disabled={disabled} themes={theme} className={className}>
+        {/* eslint-disable-next-line smarthr/a11y-input-has-name-attribute */}
+        <Checkbox {...props} disabled={disabled} type="checkbox" role="switch" ref={ref} />
+        <CheckIcon themes={theme} />
+      </Wrapper>
+    )
+  },
+)
+
+const Wrapper = styled.span<{ $disabled: Props['disabled']; themes: Theme }>`
+  ${({ $disabled, themes: { border, color, radius, shadow } }) => css`
+    position: relative;
+    display: inline-flex;
+    align-items: center;
+    border-radius: 1.25rem;
+    border: ${border.shorthand};
+    background-color: ${color.WHITE};
+
+    /* 理想的には padding: 2px; だが、box-shadow を outline で使用しているため、border と padding で2pxの疑似余白を作っている。 */
+    padding: 1px;
+
+    /* :focus-visible-within の代替 */
+    :has(:focus-visible) {
+      ${shadow.focusIndicatorStyles}
+    }
+
+    ::before,
+    ::after {
+      content: '';
+      box-sizing: border-box;
+      display: inline-block;
+      border-radius: ${radius.full};
+      border: ${border.shorthand};
+      border-color: ${color.GREY_30};
+      background-color: ${color.GREY_30};
+      block-size: 1rem;
+      inline-size: 1rem;
+      transition-property: transform, scale;
+      transition-duration: 150ms;
+      transition-timing-function: ease-out;
+    }
+
+    ::before {
+      scale: ${2 / 3};
+    }
+
+    /* ツマミが移動する余白のために存在している */
+    ::after {
+      visibility: hidden;
+    }
+
+    :has(:checked) {
+      border-color: ${color.MAIN};
+      background-color: ${color.MAIN};
+
+      ::before {
+        scale: unset;
+        transform: translateX(1rem);
+        border-color: ${color.BORDER};
+        background-color: ${color.WHITE};
+      }
+    }
+
+    ${$disabled &&
+    css`
+      &&& {
+        cursor: unset;
+        border-color: ${color.BORDER};
+        background-color: ${color.BORDER};
+      }
+    `}
+
+    @media (forced-colors: active) {
+      ${$disabled
+        ? css`
+            &&& {
+              border-color: GrayText;
+
+              &::before,
+              &::after {
+                border-color: GrayText;
+              }
+            }
+          `
+        : css`
+            /* ツマミを塗りつぶす https://developer.mozilla.org/en-US/docs/Web/CSS/system-color */
+            &&&::before,
+            &&&::after {
+              background-color: ButtonBorder;
+            }
+          `}
+    }
+
+    @supports not selector(:has(+ *)) {
+      border-style: revert;
+      background-color: revert;
+
+      ::before,
+      ::after {
+        display: none;
+      }
+
+      ${$disabled &&
+      css`
+        &&& {
+          background-color: revert;
+        }
+      `}
+    }
+  `}
+`
+
+// eslint-disable-next-line smarthr/a11y-prohibit-input-placeholder, smarthr/a11y-input-has-name-attribute
+const Checkbox = styled.input`
+  position: absolute;
```

Comment:
@uknmr 
自分のエミュレーター環境ですと、タップできる領域が「隠れているチェックボックスの幅」に依存している模様でした。↓の行を追加することで領域が拡大され、タップできるようになるのを確認しました。
```suggestion
  position: absolute;
  width: 100%;
```

Reply:
手元の iOS では `inset` で広がってるんですが、エミュレーターで使われてるバージョンが古い可能性があるかもしれません。念のためコメント付きで追加します!
https://caniuse.com/mdn-css_properties_inset

Reply:
対応してみました!
ちなみに iOS のエミュレーターは何を使ってるんですか?
[ca5f4e0](https://github.com/kufu/smarthr-ui/pull/3719/commits/ca5f4e01d71e8cdc091ae101b98c42bf31fa82dc)

Reply:
@uknmr 
対応ありがとうございます〜〜！（コメント付き良さそうです）
エミュレーターはiOS16.5を使っていました。can i use見る限りはinsetにも対応しているようにも見えますが、どうして...。